Report fastlane.swift usage vs normal fastlane (ruby) usage

taquitos opened the pull request:
We want to understand how many folks are using fastlane.swift, so we need to report execution of fastlane.swift vs normal fastlane.
Also includes a tiny bit of cleanup.

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

Commit:
Report fastlane.swift usage vs normal fastlane (ruby) usage

We want to understand how many folks are using fastlane.swift, so we need to report execution of fastlane.swift vs normal fastlane.
Also includes a tiny bit of cleanup.

taquitos commented on fastlane_core/lib/fastlane_core/analytics/analytics_event_builder.rb:
```diff
@@ -2,16 +2,18 @@ module FastlaneCore
       {
         client_id: @p_hash,
-        category: :undefined,
+        category: @fastlane_client_language,
```
cc: @powerivq ^

powerivq commented:
There are very limited dimensions we could use for a Google Analytics event (category, label and value). I wish we could make the values like a map (something like "lang=ruby,another_cat=val") in the event label, so that we can group the users in any conceivable way.

taquitos commented:
@powerivq right now, we only care about ruby vs swift. After a while, we should probably remove this code anyway. I don't see us adding a bunch of analytics, so I think utilizing a category for this is ok, for now 😄

Commit:
Make category more friendly

as described in best practices for Category: https://support.google.com/analytics/answer/1033068?hl=en

joshdholtz commented on fastlane_core/lib/fastlane_core/analytics/action_launch_context.rb:
```diff
@@ -8,22 +8,22 @@ class ActionLaunchContext
       @p_hash = p_hash
       @platform = platform
-      @configuration_language = configuration_language
+      @fastlane_client_language = fastlane_client_language
```
@taquitos @powerivq Running an action directly (like `fastlane run debug`) actually has different values for `@configuration_language` / `@fastlane_client_language` before and after this change.

Before this change, `@configuration_language` had a value of `ruby` for a Swift project
After this change, `@fastlane_client_language` has a value of `swift` for a Swift project

Both **can** be right but I want to make sure that we are recording the right thing. Are we recording...
1) the client language that is initiating the action? This would be for us recording language stats per action.
or
2) the client language that the whole fastlane project is configured with? This would be for us recording language configured specifically per project and not directly what language is executing that action.

If its **1)**, I think this PR needs a change. `fastlane run debug` doesn't do anything with Swift so this will inflate the Swift analytics as the users isn't actually running anything fastlane with their `Fastfile.swift`
If its **2)**, this PR is 💯.

Lemme know if this doesn't make sense or isn't a problem 🙃

taquitos replied:
Great catch @joshdholtz! 
I think **(2)** is OK for what we're trying to do right now. I want to see how many folks have fastlane configured for their project. It would be really helpful to understand this number so we can figure out how to move fastlane.swift forward.
I had the sense that doing **(1)** would be a bit more accurate, but at the same time, a bit more involved. Due to the timing of recording the launch actions, it might mean we would need to rejigger the analytics calls a bit. I didn't want to do that this time around 😄

joshdholtz reviewed (commented):
Got a question about _what_ we are actually trying to record here 🙃 (see comment in `action_launch_context.rb`)

joshdholtz reviewed (approved):
Looks good! 👍 LGTM 💯
